src/Antiforgery/benchmarks/Microsoft.AspNetCore.Antiforgery.Benchmarks/Benchmarks/AntiforgeryBenchmarks.cs

```diff
@@ -0,0 +1,143 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Claims;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Antiforgery.Benchmarks.Benchmarks;
+
+[AspNetCoreBenchmark]
+public class AntiforgeryBenchmarks
+{
+    private IServiceProvider _serviceProvider = null!;
+    private IAntiforgery _antiforgery = null!;
+    private string _cookieName = null!;
+    private string _formFieldName = null!;
+
+    // For GetAndStoreTokens - fresh context each time
+    private HttpContext _getAndStoreTokensContext = null!;
+
+    // For ValidateRequestAsync - context with valid tokens
+    private HttpContext _validateRequestContext = null!;
+    private string _cookieToken = null!;
+    private string _requestToken = null!;
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.AddAntiforgery();
+        serviceCollection.AddLogging();
+        _serviceProvider = serviceCollection.BuildServiceProvider();
+
+        _antiforgery = _serviceProvider.GetRequiredService<IAntiforgery>();
+
+        // Get the actual cookie and form field names from options
+        var options = _serviceProvider.GetRequiredService<IOptions<AntiforgeryOptions>>().Value;
+        _cookieName = options.Cookie.Name!;
+        _formFieldName = options.FormFieldName;
+
+        // Setup context for GetAndStoreTokens (no existing tokens)
+        _getAndStoreTokensContext = CreateHttpContext();
+
+        // Generate tokens for validation benchmark
+        var tokenContext = CreateHttpContext();
+        var tokenSet = _antiforgery.GetAndStoreTokens(tokenContext);
+        _cookieToken = tokenSet.CookieToken!;
+        _requestToken = tokenSet.RequestToken!;
+
+        // Setup context for ValidateRequestAsync (with valid tokens)
+        _validateRequestContext = CreateHttpContextWithTokens(_cookieToken, _requestToken);
+    }
+
+    [IterationSetup(Target = nameof(GetAndStoreTokens))]
+    public void SetupGetAndStoreTokens()
+    {
+        // Create a fresh context for each iteration to simulate real-world usage
+        _getAndStoreTokensContext = CreateHttpContext();
```

Comment:
I know this gets excluded from the measurement, but it does create GC pressure. Could we get more accurate numbers by resetting the HttpContext between iterations?